CONTRIBUTING.md

```diff
@@ -5,13 +5,55 @@ email, or any other method with the owners of this repository before making a ch
 
 Please note we have a [code of conduct](CODE_OF_CONDUCT.md) that we ask you to follow in all your interactions with the project.
 
+**IMPORTANT: Please do not create a Pull Request without creating an issue first.**
+
+*Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request.*
+
+Thank you for your pull request. Please provide a description above and review
+the requirements below.
+
 ## Pull Request Process
 
+0. Check out [Pull Request Checklist](#pull-request-checklist), ensure you have fulfilled each step.
 1. Ensure any install or build dependencies are removed before the end of the layer when doing a
    build.
-2. Update the README.md with details of changes to the interface, this includes new environment
+2. Please ensure the [README](README.md) and [DOCS](./DOCS.md) are up-to-date with details of changes to the interface, this includes new environment.
```

Comment:
What do you mean by these?
- interface
- environment

Reply:
Good points, I should fix those and I will. What I meant "command-line interface", so any changes in parameters and the last one should be "environment variables".